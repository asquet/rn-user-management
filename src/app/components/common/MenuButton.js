import React from 'react';
import Icon from 'react-native-vector-icons/EvilIcons';
import { TouchableHighlight, View } from 'react-native';
import Menu from '../Menu';

function MenuButton({ styles, menuOpen }) {
  return (
    <TouchableHighlight onPress={menuOpen} style={styles}>
      <Icon name="navicon" size={24} />
    </TouchableHighlight>
  );
}

MenuButton.propTypes = {
  styles: View.propTypes.style,
  menuOpen: React.PropTypes.func.isRequired,
};
MenuButton.defaultProps = {
  styles: null,
};

export default Menu.withMenuAwareness(MenuButton);
