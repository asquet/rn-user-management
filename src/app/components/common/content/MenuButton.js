import React from 'react';
import Icon from 'react-native-vector-icons/EvilIcons';
import { TouchableOpacity, View } from 'react-native';
import { menuStyles } from '../../../stylesheet';

function MenuButton({ styles, menuOpen }) {
  return (
    <TouchableOpacity onPress={menuOpen} style={[menuStyles.menuButton, styles]}>
      <Icon name="navicon" size={24} />
    </TouchableOpacity>
  );
}

MenuButton.propTypes = {
  styles: View.propTypes.style,
  menuOpen: React.PropTypes.func.isRequired,
};
MenuButton.defaultProps = {
  styles: null,
};

export default MenuButton;
