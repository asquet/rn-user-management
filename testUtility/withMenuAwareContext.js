import withContext from './withContext';
import Menu from '../src/app/components/Menu';

export default function withMenuAwareContext(cmp) {
  const context = {
    menuOpen: () => {
    },
    menuClose: () => {
    },
  };
  return withContext({ context, contextType: Menu.childContextTypes })(cmp);
}
