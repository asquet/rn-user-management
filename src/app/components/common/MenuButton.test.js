import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MenuButton from './MenuButton';
import withMenuAwareContext from '../../../../testUtility/withMenuAwareContext';

// Note: test renderer must be required after react-native.

describe('MenuButton', () => {
  it('renders correctly', () => {
    const Cmp = withMenuAwareContext(MenuButton);
    const tree = renderer.create(
      <Cmp />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
