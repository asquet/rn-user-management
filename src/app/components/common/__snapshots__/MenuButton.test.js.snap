// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "padding": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "EvilIcons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï„ 
  </Text>
</View>
`;
